packages/valory/skills/decision_maker_abci/behaviours/sell_outcome_token.py

```diff
@@ -0,0 +1,215 @@
+from packages.valory.skills.decision_maker_abci.states.sell_outcome_token import (
+    SellOutcomeTokenRound,
+)
+
+
+class SellTokenBehaviour(DecisionMakerBaseBehaviour):
+    """A behaviour in which the agents sell a token."""
+
+    matching_round = SellOutcomeTokenRound
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize the sell token behaviour."""
+        super().__init__(**kwargs)
+        self.sell_amount = 0
+        self.outcome_token_balance = 0
+        self.return_amount = 0
+
+    @property
+    def market_maker_contract_address(self) -> str:
+        """Get the contract address of the market maker on which the service is going to place the bet."""
+        return self.sampled_bet.id
+
+    @property
+    def outcome_index(self) -> int:
+        """Get the index of the outcome for which the service is going to sell token."""
+        return cast(int, self.synchronized_data.vote)
+
+    @property
+    def current_condition_id(self) -> PublicId:
+        """Get the current condition id."""
+        return self.shared_state.current_condition_id
+
+    def check_outcome_token_balance(self) -> WaitableConditionType:
+        """Check the safe's balance for a particular outcome token."""
+        response_msg = yield from self.contract_interact(
+            performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,  # type: ignore
+            contract_address=self.params.conditional_tokens_address,
+            contract_public_id=ConditionalTokensContract.contract_id,
+            contract_callable="balanceOf",
+            data_key="data",
+            placeholder="outcome_token_balance",
+            id=self.current_condition_id,
+            owner=self.synchronized_data.safe_contract_address,
+        )
+        return response_msg
+
+    def _build_approval_tx(self) -> WaitableConditionType:
+        """Build an ERC20 approve transaction."""
+        response_msg = yield from self.get_contract_api_response(
+            performative=ContractApiMessage.Performative.GET_STATE,  # type: ignore
+            contract_address=self.collateral_token,
+            contract_id=str(ERC20.contract_id),
+            contract_callable="build_approval_tx",
+            spender=self.market_maker_contract_address,
+            amount=self.return_amount,
+        )
+
+        if response_msg.performative != ContractApiMessage.Performative.STATE:
+            self.context.logger.info(f"Could not build approval tx: {response_msg}")
+            return False
+
+        approval_data = response_msg.state.body.get("data")
+        if approval_data is None:
+            self.context.logger.info(f"Could not build approval tx: {response_msg}")
+            return False
+
+        batch = MultisendBatch(
+            to=self.market_maker_contract_address,
+            data=HexBytes(approval_data),
+        )
+        self.multisend_batches.append(batch)
+        return True
```

Comment:
You could move the `_build_approval_tx` from the bet placement behaviour to the base and reuse it instead.

Reply:
done